contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java

```diff
@@ -80,8 +86,21 @@ public void start(Workflow workflow, Task task, WorkflowExecutor executor) throw
 			task.setStatus(Status.FAILED);
 			return;
 		} else {
-			if (url != null) {
-				input.setUri(url + input.getUri());
+			final String uri = input.getUri();
+			url = StringUtils.defaultIfEmpty(url, "");
+
+			if (uri.startsWith("/")) {
+				input.setUri(url + uri);
+			} else {
+				// https://jira.d3nw.com/browse/ONECOND-837
+				// Parse URI, extract the path, and append it to url
+				try {
+					URL tmp = new URL(uri);
+					input.setUri(url + tmp.getPath());
+				} catch (MalformedURLException e) {
+					logger.error("Unable to parse URL: " + uri, e);
+					// Oleksiy: In this case, the code will continue with the original URI value. Will that cause any issues?
```

Comment:
The uri must be absolute. So, yes, it will fail if URL.getPath throws ex eption

Reply:
Should we return in that case?

Reply:
Updated in 5942fde1 to throw an exception instead.